Added wrapper for set-icon

lalzy opened the pull request:
The window-frame icon that is.

lalzy committed:
Created set-icon so you can easily set window-icon

This was missing a wrapper.

elliottslaughter commented on lispbuilder-sdl/sdl/image.lisp:
```diff
@@ -292,3 +292,19 @@ Parameters supported are the same as those for [LOAD-IMAGE](#load-image) and `CO
+
+Returns the icon as `FOREIGN-OBJECT`.
+
+if `MASK` is passed, it must be an `UINT8` foreign pointer to an 32x32 (128) bitmask array. 
+       - Note; had issues with 128, so suggest 255 just to be safe. Make no difference since 255 is allocated anyway."
```
Since I got confused initially, maybe clarify this is 1024 bits (128 bytes).

Also, to be clear: is the "issue" described in this comment when the *user* passes 128 bytes, or when the code below (the contents of this PR) modified to allocate 128 bytes?

lalzy replied:
~~No, it's 1 byte (255 bits), not 255 bytes. 'theoretically' 128 bytes should be enough, but in my testing, when using less than about 200 bits, this happens(images). Why? I have no idea. but since the lowest amount of memory is 1 byte, setting the remaining bits to fix it = doesn't really matter, does it?~~

This is without altered code (i.e; mask is the same). So the mask is being applied unregularly, which from my experimenting with the mouse-cursor, shows the mask expected is too small, hence why I just went with 255 (so filling out the byte).

![transparency at wrong spots2](https://github.com/user-attachments/assets/d3013425-9f5b-4124-a13e-359262fa3d62)
![transparency at wrong spots](https://github.com/user-attachments/assets/1bfb024f-f6e1-4229-a8b1-33e0271327cd)


-- edit --

Confused myself, lol. 🤦‍♂️. 

--

Yeah, should probably write that it's bytes, will do it in the morning.

lalzy replied:
Alright, did some checking, and found the issue.... I (was very certain) I made the image 32x32.bmp in mspaint,  however, somehow it became 32x41........

Will fix that tomorrow and set it to 128bytes as it's supposed to be...

lalzy committed:
Better commenting, reverted to 128 byte, fixed pathnames.

I had issues with 128bytes because I wasn't passing an 128 byte image
to begin with (somehow 32x32 turnd into 32x41)... So that works.

Commenting to make it clearer both what is being done (manual creation
of an 128 byte mask, with all bits set, a.k.a full visibility).

And lastly convert source to namestring since sdl-load-bmp only
accepts a string, and I didn't think about it until I tested with namestrings...

lalzy commented:
Any chance to have this looked at (and hopefully merged) @Balooga?

Balooga commented:
@lalzy Looked at and merged!
